backend/inspirehep/migrator/api.py

```diff
@@ -25,39 +25,40 @@ def continuous_migration():
     """Task to continuously migrate what is pushed up by Legacy."""
     # XXX: temp redis url when we use continuous migration in kb8s
     redis_url = current_app.config.get("MIGRATION_REDIS_URL")
-    queue = "continuous_migration"
-
     if redis_url is None:
         redis_url = current_app.config.get("CACHE_REDIS_URL")
 
-    LOGGER.debug("Connected to REDIS", redis_url=redis_url, queue=queue)
+    LOGGER.debug("Connected to REDIS", redis_url=redis_url)
 
     r = StrictRedis.from_url(redis_url)
-    lock = Lock(r, queue, expire=120, auto_renewal=True)
-
-    if lock.acquire(blocking=False):
-        try:
-            migrated_records = None
-            num_of_records = r.llen("legacy_records")
-            LOGGER.info("Starting migration of records.", records_total=num_of_records)
-
-            while r.llen("legacy_records"):
-                raw_record = r.lrange("legacy_records", 0, 0)
-                if raw_record:
-                    migrated_records = insert_into_mirror(
-                        [zlib.decompress(raw_record[0])]
-                    )
-                    migrated_records_len = len(migrated_records)
-                    LOGGER.debug(
-                        "Finished migrating records.",
-                        records_total=migrated_records_len,
-                    )
+    lock = Lock(r, "continuous_migration", expire=120, auto_renewal=True)
+
+    message = r.lrange("legacy_records", 0, 0)
+    if not message:
+        LOGGER.debug("No records to migrate.")
+        return
+
+    if not lock.acquire(blocking=False):
+        LOGGER.info("Continuous_migration already executed. Skipping.")
+        return
+
+    try:
+        num_of_records = r.llen("legacy_records")
+        LOGGER.info("Starting migration of records.", records_total=num_of_records)
+
+        while message:
+            if message[0] == b"END":
```

Comment:
you may ignore it if you like, but we're using ``message[0]`` in multiple places, maybe assign it?